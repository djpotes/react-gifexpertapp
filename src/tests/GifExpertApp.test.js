import '@testing-library/jest-dom'

import GifExpertApp from '../GifExpertApp';
import {shallow}  from "enzyme"

describe('Pruebas del component <GifExpertApp />', () =>{

    let wrapper;
    test('Debe de mostrar el component correctamente', () => {
        wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', "Hello"]
        wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length)
    });
})