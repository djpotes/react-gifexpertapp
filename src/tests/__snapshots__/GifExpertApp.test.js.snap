// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del component <GifExpertApp /> Debe de mostrar el component correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas del component <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Hello"
      key="Hello"
    />
  </ol>
</Fragment>
`;
